pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentMessageExpiryMonitor.java

```diff
@@ -94,6 +95,32 @@ public void expireMessages(int messageTTLInSeconds) {
         }
     }
 
+    public void expireMessages(Position messagePosition) {
+        // If it's beyond last position of this topic, do nothing.
+        if (((PositionImpl) subscription.getTopic().getLastPosition()).compareTo((PositionImpl) messagePosition) < 0) {
+            return;
+        }
+        if (expirationCheckInProgressUpdater.compareAndSet(this, FALSE, TRUE)) {
+            log.info("[{}][{}] Starting message expiry check, position= {} seconds", topicName, subName,
+                    messagePosition);
+
+            cursor.asyncFindNewestMatching(ManagedCursor.FindPositionConstraint.SearchActiveEntries, entry -> {
+                try {
+                    // If given position larger than entry position.
+                    return ((PositionImpl) entry.getPosition()).compareTo((PositionImpl) messagePosition) <= 0;
+                } finally {
+                    entry.release();
+                }
+            }, this, null);
+        } else {
+            if (log.isDebugEnabled()) {
```

Comment:
I called an API, and the server returned success, but the final result may be not success at all. This looks like a bug. So, is it better to throw an exception or let client know that broker is busy?  This is just a suggestion

Reply:
I guess it's because originally it's called internally to expire message so it's fine to just fail silently if there's already an ongoing expiration operation. 
But you're right when we expose it as external API, the cli shouldn't throw no exception but internally it skip the operation.
However to fix the behavior will require to make changes in other places that use this `expireMessages` method which will change the scope of this pr, so I think we better take care of that in a separate issue.